Makefile

```diff
@@ -34,27 +27,35 @@ nofuse:
 
 test: test_expensive
 
-test_short: build test_go_short test_sharness_short
+test_short: test_go_short test_sharness_short
 
-test_expensive: build test_go_expensive test_sharness_expensive
+test_expensive: test_go_expensive test_sharness_expensive
 
 test_3node:
 	cd test/3nodetest && make
 
-test_go_short:
-	$(go_test) -test.short ./...
+ifeq ($(TEST_NO_FUSE),1)
+GO_TEST_FLAGS = -tags nofuse
+endif
+
+test_go_short: GO_TEST_FLAGS += -test.short
+test_go_short: test_go
+
+test_go_expensive: test_go
+
+test_go_race: GO_TEST_FLAGS += -race
+test_go_race: test_go
 
-test_go_expensive:
-	$(go_test) ./...
+test_go: build
+	go test $(GO_TEST_FLAGS) ./...
 
-test_go_race:
-	$(go_test) ./... -race
+test_sharness_short: test_sharness
 
-test_sharness_short:
-	cd test/sharness/ && make
+test_sharness_expensive: SHARNESS_MAKE_FLAGS += TEST_EXPENSIVE=1
```

Comment:
i prefer explicitness in these targets. adding flags like that is errorprone in my opinion.